### ISSUE TITLE
Kubernetes V2 Recreate Strategy should act as an Upsert and use 'kubectl create' for items that don't exist.

### ISSUE THREAD

**Issue by clanesf** [spinnaker/spinnaker#5148]
### Issue Summary:
When using a `recreate` strategy to deploy ConfigMaps that are expected to be greater than the 256Kb kubectl `last-applied-configuration` annotation limit, the current implementation is such that you must *first* deploy the ConfigMap using a standard strategy and then update the strategy to recreate. 

This causes a workflow where an end user must deploy at least two versions of a helm chart instead of one. Ideally the `recreate` strategy would act as an upsert, meaning if the item doesn't exist it would run `kubectl create` and if it does exist then it would run `kubectl replace` for the resource. 

### Cloud Provider(s):
Kubernetes V2

### Feature Area:
Deploy Manifest

**Comment by spinnakerbot** [comment]
This issue hasn't been updated in 45 days, so we are tagging it as 'stale'. If you want to remove this label, comment:

> @spinnakerbot remove-label stale

**Comment by ezimanyi** [comment]
@clanesf : I just took a look at how this works, and tried myself, and it looks like what happens when you set the strategy to `recreate` is:
* We first try to delete any existing object with that name (with `kubectl delete`); this will fail if it doesn't exist, but we ignore the failure
* We then deploy the new object using `kubectl apply`

I was able to get this to work---I deployed an object that did not yet exist using the `recreate` strategy and it was sucesfully deployed.  That being said, given that the `recreate` strategy ultimately uses `kubectl apply` and so does add the `last-applied-configuration` annotation.

Did you mean to report this issue for the `replace` strategy, which I think is what you want to avoid adding the `last-applied-configuration` annotation?  In that case, we are calling out to `kubectl replace` which will fail if the resource doesn't exist already.  It does seem like we should find a better way of supporting this use case in Spinnaker, but I want to make sure I'm understanding the problem first.
